### ISSUE TITLE
Can't send

### ISSUE THREAD

**Issue by airtoxin** [sowawa/fluent-plugin-slack#16]
I try to send logs to my slack team using this but it won't send any logs.

```
2015-05-14 21:18:04 +0900 [info]: out_slack: post_message {:channel=>"#logs", :text=>"oijwef\n", :mrkdwn=>true, :link_names=>true}
2015-05-14 21:18:04 +0900 [warn]: temporarily failed to flush the buffer. next_retry=2015-05-14 21:20:03 +0900 error_class="NameError" error="uninitialized constant Net::OpenTimeout" instance=70183716234940
  2015-05-14 21:18:04 +0900 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluent-plugin-slack-0.6.0/lib/fluent/plugin/out_slack.rb:131:in `rescue in write'
  2015-05-14 21:18:04 +0900 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluent-plugin-slack-0.6.0/lib/fluent/plugin/out_slack.rb:128:in `write'
  2015-05-14 21:18:04 +0900 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluentd-0.10.55/lib/fluent/buffer.rb:296:in `write_chunk'
  2015-05-14 21:18:04 +0900 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluentd-0.10.55/lib/fluent/buffer.rb:276:in `pop'
  2015-05-14 21:18:04 +0900 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluentd-0.10.55/lib/fluent/output.rb:311:in `try_flush'
  2015-05-14 21:18:04 +0900 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluentd-0.10.55/lib/fluent/output.rb:132:in `run'
2015-05-14 21:18:06 +0900 applicationlog.testtest: {"message":"oijwef"}
```

my config

```
<match applicationlog.testtest>
type copy
<store>
  type stdout
</store>
<store>
  type slack
  webhook_url http://hooks.slack.com/services/xxxxxx/xxxxx/xxxxxxxxx
  channel logs
  flush_interval 1s
</store>
</match>
```

**Comment by sonots** [comment]
It is wierd. `out_slack.rb` does `require` `slack_client.rb` and, it does `require` `net/http` at https://github.com/sowawa/fluent-plugin-slack/blob/34ba9b2d8ffa946a135918826a411cbd304e8037/lib/fluent/plugin/slack_client.rb#L2. So, `Net::OpenTimeout` must be defined. 

In my environment, irb tells like follows:
```
irb> Net::OpenTimeout
NameError: uninitialized constant Net
irb> require 'net/http'
irb> Net::OpenTimeout
Net::OpenTimeout
```

Could you tell me your td-agent version and its ruby version?

**Comment by airtoxin** [comment]
I use system ruby 2.1.2 and td-agent 0.10.53
But fluentd use ruby 1.9.1.
fluentd's irb says

```
irb(main):001:0> Net::OpenTimeout
NameError: uninitialized constant Net
	from (irb):1
	from /usr/lib64/fluent/ruby/bin/irb:12:in `<main>'
irb(main):002:0> require 'net/http'
=> true
irb(main):003:0> Net::OpenTimeout
NameError: uninitialized constant Net::OpenTimeout
	from (irb):3
	from /usr/lib64/fluent/ruby/bin/irb:12:in `<main>'
```

And another try, I insert some debug logs in `slack_client.rb`

```ruby
      def post(endpoint, params)
log.info "@endpoint: ", endpoint
        http = proxy_class.new(endpoint.host, endpoint.port)
        http.use_ssl = (endpoint.scheme == 'https')
        http.verify_mode = OpenSSL::SSL::VERIFY_NONE
        http.set_debug_output(debug_dev) if debug_dev
log.info "@http: ", http

        req = Net::HTTP::Post.new(endpoint)
log.info "@req: ", req
        req['Host'] = endpoint.host
        req['Accept'] = 'application/json; charset=utf-8'
        req['User-Agent'] = 'fluent-plugin-slack'
        req.body = encode_body(params)

        res = http.request(req)
        response_check(res, params)
      end
```

and got a new log

```
2015-05-15 10:02:35 +0900 [info]: out_slack: post_message {:channel=>"#logs", :text=>"oooooooooooooo\n", :mrkdwn=>true, :link_names=>true}
2015-05-15 10:02:35 +0900 [info]: @endpoint: https://hooks.slack.com/services/xxxxx/xxxxx/xxxxxxxxxx
2015-05-15 10:02:35 +0900 [info]: @http: #<Net::HTTP:0x007f726ffbb268>
2015-05-15 10:02:35 +0900 [warn]: temporarily failed to flush the buffer. next_retry=2015-05-15 10:07:19 +0900 error_class="NameError" error="uninitialized constant Net::OpenTimeout" instance=70064779390100
  2015-05-15 10:02:35 +0900 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluent-plugin-slack-0.6.0/lib/fluent/plugin/out_slack.rb:131:in `rescue in write'
  2015-05-15 10:02:35 +0900 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluent-plugin-slack-0.6.0/lib/fluent/plugin/out_slack.rb:128:in `write'
  2015-05-15 10:02:35 +0900 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluentd-0.10.53/lib/fluent/buffer.rb:296:in `write_chunk'
  2015-05-15 10:02:35 +0900 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluentd-0.10.53/lib/fluent/buffer.rb:276:in `pop'
  2015-05-15 10:02:35 +0900 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluentd-0.10.53/lib/fluent/output.rb:311:in `try_flush'
  2015-05-15 10:02:35 +0900 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluentd-0.10.53/lib/fluent/output.rb:132:in `run'
```

no req output so `Net::HTTP::Post.new(endpoint)` failed.

**Comment by sonots** [comment]
@airtoxin I fixed fluent-plugin-slack to work with ruby 1.9.3. Could you try the latest version? It could work with 1.9.1 possibly although I won't support ruby 1.9.1 officially. 1.9.1, even 1.9.3 are end of life already.

**Comment by airtoxin** [comment]
@sonots It works!
And I try to update td-agent newer one. Very thank you!
